server/app/services/applicant/ApplicantService.java

```diff
@@ -429,6 +429,28 @@ private void maybeNotifyApplicant(long applicantId, long applicationId, String p
     }
   }
 
+  /** Return the name of the given applicant id. */
+  public CompletionStage<Optional<String>> getNameOrEmail(long applicantId) {
+    return userRepository
```

Comment:
If this method returns an empty optional, the UI says "Logged in as Guest"

Reply:
Will add a test

Reply:
Aha. I think I was misunderstanding things from the PR description.

Is this line from the description stating the pre-PR behavior?
> Also, when a user account is created without a name, shows "Logged in as " instead of "Logged in as Guest" until the Name cannonical question is completed.

Reply:
Correct

Reply:
OOOOOOHHHHH.  It shoudl be Logged in as `<email>`" in the description. Markdown ate my `<email>`

Reply:
s/"Logged in as " "Logged in as `<guest>".  Markdown eats things inside angle brackets